migration/process-migration/src/main/java/io/camunda/migration/process/adapter/os/OpensearchAdapter.java

```diff
@@ -0,0 +1,246 @@
+
+  private static final Logger LOG = LoggerFactory.getLogger(OpensearchAdapter.class);
+  private final ProcessMigrationProperties properties;
+  private final OpenSearchClient client;
+  private final MigrationRepositoryIndex migrationRepositoryIndex;
+  private final ProcessIndex processIndex;
+  private final ImportPositionIndex importPositionIndex;
+
+  public OpensearchAdapter(
+      final ProcessMigrationProperties properties,
+      final ConnectConfiguration connectConfiguration) {
+    this.properties = properties;
+    migrationRepositoryIndex =
+        new MigrationRepositoryIndex(connectConfiguration.getIndexPrefix(), false);
+    processIndex = new ProcessIndex(connectConfiguration.getIndexPrefix(), false);
+    importPositionIndex = new ImportPositionIndex(connectConfiguration.getIndexPrefix(), false);
+    client = new OpensearchConnector(connectConfiguration).createClient();
+  }
+
+  @Override
+  public String migrate(final List<ProcessEntity> entities) throws MigrationException {
+    final BulkRequest.Builder bulkRequest = new BulkRequest.Builder();
+    final var idList = entities.stream().map(ProcessEntity::getId).toList();
+    entities.forEach(e -> migrateEntity(e, bulkRequest));
+
+    final BulkResponse response;
+    try {
+      response =
+          doWithRetry(
+              "Migrate entities %s".formatted(idList),
+              () -> client.bulk(bulkRequest.build()),
+              (res) -> res == null || res.errors() || res.items().isEmpty());
+    } catch (final Exception e) {
+      throw new MigrationException("Failed to migrate entities %s".formatted(idList), e);
+    }
+    return lastUpdatedProcessDefinition(response.items());
+  }
+
+  @Override
+  public List<ProcessEntity> nextBatch(final String processDefinitionKey) {
+    final SearchRequest request =
+        new SearchRequest.Builder()
+            .index(processIndex.getFullQualifiedName())
+            .size(properties.getBatchSize())
+            .sort(s -> s.field(f -> f.field(PROCESS_DEFINITION_KEY).order(SortOrder.Asc)))
+            .query(
+                q ->
+                    q.range(
+                        r ->
+                            r.field(PROCESS_DEFINITION_KEY)
+                                .gt(
+                                    JsonData.of(
+                                        processDefinitionKey == null ? "" : processDefinitionKey))))
+            .build();
+
+    final SearchResponse<ProcessEntity> searchResponse;
+    try {
+      searchResponse =
+          doWithRetry(
+              "Fetching next process batch",
+              () -> client.search(request, ProcessEntity.class),
+              res -> res.timedOut() || Boolean.TRUE.equals(res.terminatedEarly()));
+    } catch (final Exception e) {
+      throw new MigrationException("Failed to fetch next processes batch", e);
+    }
+    return searchResponse.hits().hits().stream().map(Hit::source).toList();
+  }
```

Comment:
Technically we could, however this would require additional dependencies to be loaded in the module. Also, generally speaking I believe that data migrations are better to be interacting with the database directly and not include intermediates. Reduce liability factors :smiling_face_with_tear: